From e9b4e5d675055cf51b87a0a1e10a60983b2dcc8c Mon Sep 17 00:00:00 2001
From: Zhibin Huang <zhibin.huang@rock-chips.com>
Date: Tue, 27 Feb 2024 12:39:49 +0800
Subject: [PATCH 301/305] misc: rk628: bt1120-2-hdmi: fix bt1120 timing cfg
 when switching the resolution

Type: Fix
Redmine ID: N/A
Associated modifications: N/A
Test: N/A

Signed-off-by: Zhibin Huang <zhibin.huang@rock-chips.com>
Change-Id: Ied0d5322338feee4f59608cd3468444ec1f4dcc7
Signed-off-by: Guochun Huang <hero.huang@rock-chips.com>
---
 drivers/misc/rk628/rk628.c              | 8 ++++++--
 drivers/misc/rk628/rk628_post_process.c | 8 ++++++++
 2 files changed, 14 insertions(+), 2 deletions(-)

diff --git a/drivers/misc/rk628/rk628.c b/drivers/misc/rk628/rk628.c
index a40cee4446d98b..4b6e5c93879534 100644
--- a/drivers/misc/rk628/rk628.c
+++ b/drivers/misc/rk628/rk628.c
@@ -441,8 +441,12 @@ static void rk628_display_enable(struct rk628 *rk628)
 	if (rk628_input_is_rgb(rk628))
 		rk628_rgb_rx_enable(rk628);
 
-	if (rk628_input_is_bt1120(rk628))
-		rk628_bt1120_rx_enable(rk628);
+	/*
+	 * bt1120 needs to configure the timing register, but hdmitx will modify
+	 * the timing as needed, so the bt1120 enable process is moved to the
+	 * configuration of post_process (function rk628_post_process_enable in
+	 * rk628_post_process.c)
+	 */
 
 	if (rk628_output_is_rgb(rk628))
 		rk628_rgb_tx_enable(rk628);
diff --git a/drivers/misc/rk628/rk628_post_process.c b/drivers/misc/rk628/rk628_post_process.c
index 04d452f7870564..4cbeb48450c006 100644
--- a/drivers/misc/rk628/rk628_post_process.c
+++ b/drivers/misc/rk628/rk628_post_process.c
@@ -9,6 +9,8 @@
 #include "rk628.h"
 #include "rk628_config.h"
 #include "rk628_cru.h"
+#include "rk628_post_process.h"
+#include "rk628_rgb.h"
 
 #define PQ_CSC_HUE_TABLE_NUM			256
 #define PQ_CSC_MODE_COEF_COMMENT_LEN		32
@@ -1624,6 +1626,12 @@ static void rk628_post_process_csc(struct rk628 *rk628)
 
 void rk628_post_process_enable(struct rk628 *rk628)
 {
+	/*
+	 * bt1120 needs to configure the timing register, but hdmitx will modify
+	 * the timing as needed, so the bt1120 enable process is moved here.
+	 */
+	if (rk628_input_is_bt1120(rk628))
+		rk628_bt1120_rx_enable(rk628);
 	rk628_post_process_csc(rk628);
 	rk628_i2c_write(rk628, GRF_SCALER_CON0, SCL_EN(1));
 }
-- 
2.17.1

